### PR TITLE
Version 1.2

### PR DESCRIPTION
**Pull request by QuintinWillison**
Creating this PR now, well in advance of us actually merging version 1.2 changes in to `master`, as much to obtain a deployment preview.

### CODE REVIEW

**Comment by SimonWoolf** [comment]
Is anything holding us up from releasing (at least the initial version of) the 1.2 spec? Client library developers have [now started to implement](https://github.com/ably/ably-cocoa/pull/976) the 1.2 spec, they'd probably find it helpful to refer to it at https://docs.ably.io/client-lib-development-guide/features/ rather than looking at raw textile in github. (The features spec on docs.ably.io is versioned, so if you want an earlier version you can always select it from the dropdown).

@QuintinWillison @paddybyers

File: content/client-lib-development-guide/features.textile [hunk 1]
```diff
@@ -1494,6 +1531,7 @@ class ClientOptions:
   httpMaxRetryDuration: Duration default 15s // TO3l6
   maxMessageSize: Int default 65536 // TO3l8
   maxFrameSize: Int default 524288 // TO3l8
+  plugins: Dict<PluginType, Plugin> // TO3p
```
Comment (QuintinWillison) [inline content/client-lib-development-guide/features.textile]: Comment is incorrect. Should be `TO3o`.

Comment (QuintinWillison) [reply content/client-lib-development-guide/features.textile]: https://github.com/ably/docs/pull/796/commits/dfaed3d05b2eac3d71258ba37ac21b2864afddde

**Comment by QuintinWillison** [comment]
@paddybyers, I've requested your approval please. The next step is that I land this on `master` and then I'll create one or more new PRs containing the work to make 1.2 the primary, upgrade/copy 1.1 bits / diffs / etc..

All being tracked under [FEA-379](https://ably.atlassian.net/browse/FEA-379).

FYI, @mattheworiordan, @jamienewcomb

**Pull request merged**